--- .venv/lib/python3.9/site-packages/kfac_jax/_src/optimizer.py	2022-07-14 12:48:51.387431181 +0100
+++ .venv/lib/python3.9/site-packages/kfac_jax/_src/optimizer_new.py	2022-07-14 12:48:39.959253562 +0100
@@ -304,7 +304,8 @@
     self._value_func_has_rng = value_func_has_rng
     self._value_func: ValueFunc = convert_value_and_grad_to_value_func(
         value_and_grad_func,
-        has_aux=value_func_has_aux,
+        # CHANGE
+        has_aux=value_func_has_aux or value_func_has_state,
     )
     self._l2_reg = jnp.asarray(l2_reg)
     self._use_adaptive_learning_rate = use_adaptive_learning_rate


--- .venv/lib/python3.9/site-packages/kfac_jax/_src/loss_functions.py	2023-03-14 16:48:58.069993555 +0000
+++ .venv/lib/python3.9/site-packages/kfac_jax/_src/loss_functions_new.py	2023-03-14 16:49:19.154369163 +0000
@@ -22,6 +22,7 @@
 
 from kfac_jax._src import layers_and_loss_tags as tags
 from kfac_jax._src import utils
+from kfac_jax._src.utils import types
 
 
 Array = chex.Array
@@ -43,7 +44,7 @@
       weight: The relative weight attributed to the loss.
     """
     if not isinstance(weight, (int, float)):
-      if not isinstance(weight, Array) or weight.size > 1:
+      if not types.is_array_instance(weight) or weight.size > 1:
         raise ValueError("`weight` must be a scalar value.")
     super().__init__()
     self._weight = weight

